kge/model/kge_model.py

```diff
@@ -393,23 +398,24 @@ def load_from_checkpoint(
         appended to) in the checkpoint's folder.
 
         """
+        if type(checkpoint) is str:
+            checkpoint = load_checkpoint(checkpoint, device=device)
 
-        checkpoint = torch.load(filename, map_location=device)
+        config = Config.load_from(checkpoint["config"], config)
 
-        original_config = checkpoint["config"]
-        config = Config()  # round trip to handle deprecated configs
-        config.load_options(original_config.options)
-        config.set("job.device", device)
         if use_tmp_log_folder:
             import tempfile
 
             config.log_folder = tempfile.mkdtemp(prefix="kge-")
         else:
-            config.log_folder = os.path.dirname(filename)
+            config.log_folder = checkpoint["folder"]
             if not config.log_folder:
                 config.log_folder = "."
         if dataset is None:
-            dataset = Dataset.load(config, preload_data=False)
+            if checkpoint["type"] == "package":
+                dataset.load_meta(config, checkpoint["dataset_meta"], preload_data=False)
+            else:
+                dataset = Dataset.load(config, preload_data=False)
```

Comment:
Dataset should always load what's in the checkpoint (i.e., if there is meta data load it). We probably also want to put num_entities and num_relations into the checkpoint.

Reply:
Use dataset.meta

Reply:
done

Reply:
I did not yet add num_entities and num_relations.
Do we want to have that only in the packaged model, but also in the train checkpoints?

Reply:
It should always be added in save_to for the Dataset, I guess. But we do not need to call this for training checkpoints. Generally, the datasets create_from should take whatever it finds in the checkpoint.

BTW: Please resolve all conversation which are done.